File: algorithms/implementation/p07_cut_the_sticks/p7_cut_the_sticks.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var n = parseInt(readLine());
    arr = readLine().split(' ');
    arr = arr.map(Number);
    
    var done = false;
    while (!done) {
        var count = 0;
        var cutSize = 10001;
        for (var i = 0; i < n; i++) {
            if (arr[i] > 0) {
                count += 1;
                cutSize = Math.min(cutSize, arr[i]);
            }
        }
        
        if (count === 0 || cutSize === 0) {
            break;
        }
        
        console.log(count);
        
        for (var j = 0; j < n; j++) {
            arr[j] -= cutSize;
        }
    }
}

File: algorithms/implementation/p07_cut_the_sticks/p7_cut_the_sticks.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int arr[] = new int[n];
        for(int arr_i=0; arr_i < n; arr_i++){
            arr[arr_i] = in.nextInt();
        }
        
        boolean done = false;
        while (!done) {
            // Count the positive integers and establish the cut size
            int count = 0;
            int cutSize = 10001;
            for (int i = 0; i < n; i++) {
                if (arr[i] > 0) {
                    count += 1;
                    cutSize = Math.min(cutSize, arr[i]);
                }
            }
            
            if (count == 0 || cutSize == 0) {
                break;
            }
            
            System.out.println(count);
            
            for (int i = 0; i < n; i++) {
                arr[i] -= cutSize;
            }
        }
    }
}
